tests/_data/models.py

```diff
@@ -1047,6 +1048,18 @@ def get_bom_with_multiple_licenses() -> Bom:
     )
 
 
+def get_bom_with_tools_with_component_and_service() -> Bom:
+    return _make_bom(
+        metadata=BomMetaData(
+            tools=ToolsRepository(
+                components=[Component(type=ComponentType.APPLICATION, author='adobe',
+                                      name='test-component', version='1.2.3', bom_ref='my-component')],
+                services=[Service(name='test-service', bom_ref='my-service')]
+            )
+        )
+    )
+
+
```

Comment:
additional test case: 
```py

def get_bom_with_tools_with_component_and_service_and_tools_migrate() -> Bom:
    tools = ToolsRepository()
    tcomp = tools.components
    tserv = tools.services
    ttools = tools.tools
    tcomp.add(Component(type=ComponentType.APPLICATION, name='test-component', version='1.2.3'))
    tserv.add(Service(name='test-service', bom_ref='my-service'))
    ttools.add(Tool(name='test-tool', version='1.33.7'))
    return _make_bom(metadata=BomMetaData(tools=tools))
```

Reply:
OK, I will need to think that through. As currently coded, the logic says "use components and services if schema > 1.5, otherwise use tools." See https://github.com/CycloneDX/cyclonedx-python-lib/pull/635/files#diff-739ca19bccb83b0c9db07ee076f0c86d6b7d2fbd76c63a0c5e3fd88f047b6706R337